app/controllers/user.js

```diff
@@ -0,0 +1,174 @@
+const model = require('../models');
+
+const secret = process.env.SECRET_TOKEN || 'secret';
+
+// To create text-friendly user details
+const formattedUser = (user) => {
+  const userDetails = {
+    id: user.id,
+    userName: user.userName,
+    firstName: user.firstName,
+    lastName: user.lastName,
+    email: user.email,
+    RoleId: user.RoleId,
+    createdAt: user.createdAt,
+    updatedAt: user.updatedAt
+  };
+  return userDetails;
+};
+
+/**
+ * Class UsersController
+ * To handle routing logic for documents route
+ */
+class UsersController {
+
+  /**
+   * Method getUsers to obtain all users
+   * @param {object} request - request object
+   * @param {object} response - response object
+   * @returns {Object} response object
+   */
+  static getUsers(request, response) {
+    model.User.findAll({
+      attributes: [
+        'id',
+        'userName',
+        'firstName',
+        'lastName',
+        'email',
+        'RoleId',
+        'createdAt',
+        'updatedAt'
+      ]
+    }).then((users) => {
+      return response.status(202)
+        .send(users);
+    });
+  }
+
+  /**
+   * Method createUser to create a user
+   * @param {object} request - request object
+   * @param {object} response - response object
+   * @returns {Object} - response object
+   */
+  static createUser(request, response) {
+    model.User.findOne({ where: { email: request.body.email } })
+      .then((foundUser) => {
+        if (foundUser) return response.status(409)
+            .send({ message: `Unable to create user. This mail:
+              ${request.body.email} is already in use` });
+
+        model.User.create(request.body)
+          .then((newUser) => {
+            const token = jwt.sign({
+              UserId: newUser.id,
+              RoleId: newUser.RoleId
+            }, secret, { expiresIn: '2 days' });
+
+            newUser = formattedUser(newUser);
+            return response.status(201)
+              .send({ newUser, token, expiresIn: '2 days' });
+          })
+          .catch((error) => {
+            return response.status(400)
+              .send(error.errors);
+          });
+      });
+  }
+
+  /**
+   * Method getUser to get a single user
+   * @param {object} request - request object
+   * @param {object} response - response object
+   * @returns {Object} - response object
+   */
+  static getUser(request, response) {
+    model.User.findById(request.params.id)
+      .then((foundUser) => {
+        if (!foundUser) return response.status(404)
+          .send({ message: `Ǹo user with id: ${request.params.id}` });
+
+        foundUser = formattedUser(foundUser);
+        return response.send(foundUser);
+      });
+  }
+
+  /**
+   * Method updateUser
+   * @param {object} request - request object
+   * @param {object} response - response object
+   * @returns {object} - response object
+   */
+  static updateUser(request, response) {
+    model.User.findById(request.params.id)
```

Comment:
Why not just extract this portion to another method e.g `isUserExisting` that returns true or false cos I can see that you're using it in a couple of places